src/parser/metrics/Complexity.ts

```diff
@@ -44,12 +44,94 @@ export class Complexity implements Metric {
+         */
+        const withDefaultLabelNodeType = new Set<string>();
+        const caseNodeTypes = new Set<ExpressionMetricMapping>();
+
+        for (const nodeType of allNodeTypes) {
+            if (
+                nodeType.category === NodeTypeCategory.CaseLabel &&
+                nodeType.metrics.includes(this.getName())
+            ) {
+                caseNodeTypes.add(nodeType);
+            } else if (nodeType.category === NodeTypeCategory.DefaultLabel) {
+                for (const language of nodeType.languages) {
+                    withDefaultLabelNodeType.add(language);
+                }
+            }
+        }
+
+        console.log("Language grammars with explicit default label node type: ");
+        for (const langAbbr of withDefaultLabelNodeType) {
+            console.log(langAbbr);
+        }
+
+        /**
+         * Add queries for each case label syntax node
+         */
+        for (const caseNodeType of caseNodeTypes) {
+            const haveDefaultNodeType: string[] = [];
+            const haveNoDefaultNodeType: string[] = [];
+
+            for (const languageAbbr of caseNodeType.languages) {
+                if (withDefaultLabelNodeType.has(languageAbbr)) {
+                    haveDefaultNodeType.push(languageAbbr);
+                } else {
+                    haveNoDefaultNodeType.push(languageAbbr);
+                }
+            }
+
+            // For languages which have also a default node type, always count this case node type:
+            if (haveDefaultNodeType.length > 0) {
+                this.complexityStatementsSuperSet.push(
+                    new ExpressionQueryStatement(
+                        caseNodeType.expression,
+                        haveDefaultNodeType,
+                        caseNodeType.activated_for_languages
+                    )
+                );
+            }
+
+            // Special treatment for languages which have no explicit default node type:
+            if (haveNoDefaultNodeType.length > 0) {
+                if (caseNodeType.expression == "case_statement") {
+                    // Special treatment for "case_statement" which can have more than one child,
+                    // because it also has the content of the case block as child(s).
+                    //
+                    // In this case, a case label can be differentiated from a default label by
+                    // checking if there is a child with the field name "value":
+                    this.complexityStatementsSuperSet.push(
+                        new SimpleLanguageSpecificQueryStatement(
+                            "(case_statement value: _ ) @case_statement",
+                            haveNoDefaultNodeType,
+                            caseNodeType.activated_for_languages
+                        )
+                    );
+                } else {
+                    // Check if the case label node has a named child:
```

Comment:
That means if there is a named child, it is supposed to be a case label that should be counted, otherwise it could be a default one that should be ignored?

Reply:
Yes. That works e.g. for Java.